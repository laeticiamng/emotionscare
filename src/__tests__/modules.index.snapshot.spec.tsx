import { describe, it, expect, vi } from "vitest";
import React from 'react';
vi.mock('@/components/ui/PageHeader', () => ({
  default: ({ title, subtitle }: any) =>
    React.createElement(
      'header',
      null,
      React.createElement('h1', null, title),
      React.createElement('p', null, subtitle)
    ),
}));

vi.mock('@/components/ui/card', () => ({
  Card: ({ children }: any) => React.createElement('div', null, children),
}));

vi.mock('@/components/ui/button', () => ({
  Button: (props: any) => React.createElement('a', props),
}));

vi.mock('@/components/ui/LoadingSpinner', () => {
  const LoadingSpinner = () => React.createElement('div', null, 'loading');
  return { LoadingSpinner, default: LoadingSpinner };
});
import { render } from "@testing-library/react";
import ModulesIndexPage from "@/app/modules/page";

describe("ModulesIndexPage", () => {
  it("rend la page et liste les modules", () => {
    const { container, getByText } = render(<ModulesIndexPage />);
    expect(getByText(/Modules/i)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
