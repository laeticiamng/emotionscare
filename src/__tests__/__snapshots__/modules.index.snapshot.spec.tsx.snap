// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModulesIndexPage > rend la page et liste les modules 1`] = `
<div>
  <main
    aria-label="Catalogue des modules"
  >
    <header>
      <h1>
        Modules
      </h1>
      <p>
        Explore les expériences disponibles
      </p>
    </header>
    <div
      style="display: grid; grid-template-columns: repeat(auto-fill, minmax(260px, 1fr)); gap: 12px;"
    >
      <div>
        <h2>
          FlashGlow
        </h2>
        <p>
          Accéder au module 
          flash-glow
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/flash-glow"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          AdaptiveMusic
        </h2>
        <p>
          Accéder au module 
          adaptive-music
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/adaptive-music"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          BossGrit
        </h2>
        <p>
          Accéder au module 
          boss-grit
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/boss-grit"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          BreathConstellation
        </h2>
        <p>
          Accéder au module 
          breath-constellation
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/breath-constellation"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          BubbleBeat
        </h2>
        <p>
          Accéder au module 
          bubble-beat
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/bubble-beat"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          Coach
        </h2>
        <p>
          Accéder au module 
          coach
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/coach"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          EmotionScan
        </h2>
        <p>
          Accéder au module 
          emotion-scan
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/emotion-scan"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          FlashGlowUltra
        </h2>
        <p>
          Accéder au module 
          flash-glow-ultra
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/flash-glow-ultra"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          Journal
        </h2>
        <p>
          Accéder au module 
          journal
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/journal"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          MoodMixer
        </h2>
        <p>
          Accéder au module 
          mood-mixer
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/mood-mixer"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          Scan
        </h2>
        <p>
          Accéder au module 
          scan
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/scan"
        >
          Ouvrir
        </a>
      </div>
      <div>
        <h2>
          StorySynth
        </h2>
        <p>
          Accéder au module 
          story-synth
        </p>
        <a
          data-ui="primary-cta"
          href="/modules/story-synth"
        >
          Ouvrir
        </a>
      </div>
    </div>
  </main>
</div>
`;
